pkg/agent/clean/orphan_bindings.go

```diff
@@ -194,13 +203,15 @@ func (bc *orphanBindingsCleanup) isOrphanBinding(rb rbacv1.RoleBinding) (bool, e
 	}
 
 	if hasNewLabel {
-		_, err := bc.prtbs.Get(prtbNs, prtbName, metav1.GetOptions{})
+		metadata := metav1.ObjectMeta{Name: prtbName, Namespace: prtbNs}
+		prtbs, err := bc.prtbCache.GetByIndex(prtbIndexer, rbacutil.GetRTBLabel(metadata))
```

Comment:
I don't think this will work. `prtbName` may have been trimmed and suffixed with a random string. The result of 

```
rbacutil.GetRTBLabel(metav1.ObjectMeta{Name: "reallylongn-abcd", Namespace: "ns"})
```

will be different from

```
rbacutil.GetRTBLabel(metav1.ObjectMeta{Name: "reallylongname", Namespace: "ns"})
```

The idea is the indexer should be able to look at the current label and retrieve the original object, so we want it to look like

```
prtbs, err := bc.prtbCache.GetByIndex(prtbIndexer, prtbLabel)
```

where `prtbLabel` is the actual label before the string split.

Reply:
When I add an indexer, I also specify that we want the full untouched label to be the key? And so the orphan bindings code won't need to split the label on the underscore at all?

Reply:
Yes, the label should be the key